server/controllers/userController.js

```diff
@@ -23,8 +23,7 @@ export default class UserController {
     bcrypt.hash(password, saltRounds)
       .then(hashedPassword => User.create({ username, email, password: hashedPassword, })
         .then((user) => {
-          const { id } = user;
-          const token = generateToken({ id });
+          const token = generateToken({ id: user.id, username: user.username, email: user.email });
```

Comment:
I add `username, email, and userid` into the `generateToken` function. Extra request to get this information from the server is not needed since they can be easily retrieved from the token when decoded.